### PR TITLE
style: adding erc1155 batch buy design

### PR DESCRIPTION
**Pull request by JackShort**
Implements the design for erc1155 batch buying - keep in mind this is not adding the correct behavior just the design

* this is being pushed to a feature branch btw

Screen Shot 2023-04-04 at 1 56 39 PM

### CODE REVIEW

**Review by cgkol — COMMENTED** [review]
High level (admittedly have not looked at the code at all), shall we add some tests?

**Comment by JackShort** [comment]
> High level (admittedly have not looked at the code at all), shall we add some tests?

I think the tests should be added at the end when the whole feature is implemented

**Comment by cgkol** [comment]
> > High level (admittedly have not looked at the code at all), shall we add some tests?
> 
> I think the tests should be added at the end when the whole feature is implemented - because a few things aren't accounted for here like loading the data etc.

Wait is this not finalized? Or is the next PR going to build on top of this one?

**Comment by cgkol** [comment]
btw, make sure you tag me so i get the notif 😂 @JackShort

File: src/nft/components/card/containers.tsx [hunk 1]
```diff
@@ -102,6 +110,91 @@ const ActionButton = ({
   )
 }
 
+const StyledErc1155ActionButton = styled.div<{ $disabled?: boolean }>`
```
Comment (tinaszheng) [inline src/nft/components/card/containers.tsx:113]: this is a `button` so this should just use `styled.button` 🤔 you'll get all the accessibility features associated with html buttons, and you'll also be able to just use the default `disabled` prop combined with 

```
styled.button`
   background: theme.accentAction 

   &:disabled {
      background: theme.accentActionSoft
   }
`
```

File: src/nft/components/card/icons.tsx [hunk 2]
```diff
@@ -37,17 +37,29 @@ const ListPriceRowContainer = styled(Row)`
...
   hidePrice,
 }: {
   isSelected: boolean
+  quantitySelected?: number
   marketplace?: Markets
   tokenType?: NftStandard
   listedPrice?: string
   hidePrice?: boolean
 }) => {
+  if (quantitySelected > 0) {
```
Comment (tinaszheng) [inline src/nft/components/card/icons.tsx:53]: prefer guard-style returns, like:

```
if (quantitySelected === 0) return 

return (
   <StyledMarketplaceContainer> ...
)

File: src/nft/components/card/containers.tsx [hunk 3]
```diff
@@ -102,6 +110,91 @@ const ActionButton = ({
...
+  }
+`
+
+const StyledErc1155ActionContainer = styled(ThemedText.BodySmall)`
+  display: flex;
+  background: ${({ theme }) => theme.backgroundModule};
+  color: ${({ theme }) => theme.backgroundSurface};
+  border-radius: 8px;
+  justify-content: center;
+  font-weight: 600 !important;
```
Comment (tinaszheng) [inline src/nft/components/card/containers.tsx:151]: why do you need the !important here

File: src/nft/components/card/containers.tsx [hunk 4]
```diff
@@ -30,34 +31,41 @@ const StyledDetailsContainer = styled(Column)`
...
+  will-change: transform;
+  visibility: hidden;
+
+  @media screen and (max-width: ${BREAKPOINTS.sm}px) {
+    visibility: visible;
+    bottom: 8px;
+  }
+`
+
+const ActionContainer = ({ children }: { children: ReactNode }) => (
```
Comment (tinaszheng) [inline src/nft/components/card/containers.tsx:50]: confused what this is doing - isn't this the same thing as just `StyledActionContainer` by itself?

File: src/nft/components/card/icons.tsx (hunk 2 above)
Comment (JackShort) [reply src/nft/components/card/icons.tsx:53]: this really isn't applicable we only care here if quantity selected is passed if so we render it - if not it goes to the other conditional checks below it

**Pull request closed without merging**